src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/InlineArrayAttribute.cs

```diff
@@ -1,22 +1,42 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System.ComponentModel;
-
 namespace System.Runtime.CompilerServices
 {
     /// <summary>
     /// Indicates that the instance's storage is sequentially replicated "length" times.
     /// </summary>
-    [EditorBrowsable(EditorBrowsableState.Never)]
+    /// <remarks>
+    /// <para>
+    /// This attribute can be used to annotate a <see langword="struct"/> type with a single field.
+    /// The runtime will replicate that field as many times as specified in the actual type layout.
+    /// </para>
+    /// <para>
+    /// Here's an example of how an inline array type with 8 <see cref="float"/> values can be declared:
+    /// <code lang="csharp">
+    /// [InlineArray(8)]
+    /// struct Float8InlineArray
+    /// {
+    ///     private float _value;
+    /// }
+    /// </code>
+    /// </para>
+    /// <para>
+    /// The compiler will also automatically generate an indexer property, and support casting the type
+    /// to <see cref="Span{T}"/> and <see cref="ReadOnlySpan{T}"/> for safe enumeration of all elements.
+    /// </para>
```

Comment:
Let's only talk about runtime behaviors here. There are multiple compilers, in different languages, and they won't all do this.
```suggestion
```